### DISCUSSION TITLE
Image, Containers and updating Foundry VTT

### DISCUSSION THREAD

**Discussion by mckee14** [felddy/foundryvtt-docker#208] [Q&A] [answered]
Hi All

I run this container on my Synology NAS, and I'm completely new to Docker stuff. Also, it's GUI based, so bare with me :)

I fail to understand how the "versions" of images are related to the containers, because you can specify the environment variable "FOUNDRY_VERSION" to start the container (I hope I get the terms right).

So I can basically download an image tagged with :0.8.5 and start a container from it with FOUNDRY_VERSION 0.8.5.
So far so good.

But now I can just change FOUNDRY_VERSION to 0.8.6, clear the container and it will download Foundry 0.8.6 - and it runs. On the 0.8.5 image.

Is this a good idea to do, or should you always have matching image <> Foundry Version pairing?
What if I work with the :latest image, does it automatically change/update?

I fail to grasp what is what in this whole setup. I would like to be able to just update the server without setting up a new container every time I update.

**Comment by felddy** [comment, marked as answer]
You shouldn't be setting the `FOUNDRY_VERSION` environment variable directly.  That is used internal to the image.  The Synology Docker GUI seems to expose these variables even though they aren't supposed to be used as arguments.

To make things worse, once they're exposed in the GUI they can get pinned to an old value.  

You should use the version of the container that matches the version of FoundryVTT that you want to run.  This is because there are dependency changes that need to be met (like NodeJS), as well as interfaces that change with in FoundryVTT.  Each container image is tested against the version that it supports by the CI system. 

What you are doing will work most of the time.  But it is not a supported mode of operation.  To be safe always pull the version from the registries, and don't set `FOUNDRY_VERSION` explicitly.

**Reply by mckee14** [reply]
Thank you very much to clarify this.

So it seems that i have to download a new image and re-create the container from scratch everytime I update, but like a wise man once said: it is what it is :)

**Reply by felddy** [reply]
I'm confused by "re-create the container from scratch"

You should not need to re-create the container.  Just pull the image's update, and "re-up" the container.

Maybe the Synology GUI makes this more difficult than it should be.

See: https://github.com/felddy/foundryvtt-docker#updating-your-container

**Reply by mckee14** [reply]
Yeah, it's definitely a synology problem.
In the GUI, you can "add" (download) an image, but you can not update it.

When you start a container from an image, it gets the following parameters automatically from the container, and you can't delete them (otherwise the container won't run):
![grafik](https://user-images.githubusercontent.com/62849695/120375311-f14f3c80-c31a-11eb-8ff8-d3bde56c1b29.png)

To make things worse, you can't export the container settings and re-import them for the new container based on the new image, because they contain these version information from the old image. So you have to either re-enter all the needed variables (foundry user, pw etc) every time, or merge the settings files correctly.

its not hard to do or a big deal. And I will see if the synology implementation allows to use regular docker commands via commandline. But definitely not a problem related to your docker image, and thank you for your help and your explanations.

**Reply by felddy** [reply]
Sorry.  That sounds frustrating.  FWIW, qnap and Portainer are also frustrating like that.  Hopefully you can find a way to upload a `docker-compose.yml` or something to short circuit the GUI.  

You're welcome for the help.  Good luck.

**Reply by mckee14** [reply]
Oh my, now that I understood the problem, I was able to check in the synology community. Turns out, the UX is just bad...
you CAN pull a new image, as long as the tag is the same. You have to look for it in the registry, choose to install it, and then choose the same tag. that way, the existing image gets overwritten. Then you "clear" the container, which basically does a re-up...

way to go synology, a little "pull" or "update" button would go a long way here...
